### PR TITLE
doc-gen: migrate scalar functions (array) documentation 1/3

### PR DESCRIPTION
**Pull request by Chen-Yuan-Lai**
## Which issue does this PR close?

Closes #13671 .

## What changes are included in this PR?

As discussed in #13671  , this PR will migrate the parts of builtin array functions documentation that currently support migration.

### CODE REVIEW

**Comment by Omega359** [comment]
Can you take a look at the Clippy failure? Seems to be just an unused import

**Review by comphead — APPROVED** [review]
thanks @Chen-Yuan-Lai I think we got a lot fixed in the doc with this PR

**Review by Omega359 — APPROVED** [review]
Very nice to see all the documentation fixes

File: docs/source/user-guide/sql/scalar_functions.md [hunk 1]
```diff
@@ -2907,26 +2906,26 @@ array_has(array, element)
...
 #### Example
 
 ```sql
-> select array_has([1, 2, 3], 2);
-+-----------------------------+
-| array_has(List([1,2,3]), 2) |
-+-----------------------------+
-| true                        |
-+-----------------------------+
+> select array_has_any([1, 2, 3], [3, 4]);
```
Comment (alamb) [inline docs/source/user-guide/sql/scalar_functions.md:2923]: Agree -- thank you @Chen-Yuan-Lai for your diligence while porting the code to also fix the docs

**Comment by alamb** [comment]
Thanks @Chen-Yuan-Lai  and @comphead and @Omega359

**Pull request merged**